core/src/main/java/org/kiwix/kiwixmobile/core/search/viewmodel/State.kt

```diff
@@ -29,11 +29,27 @@ sealed class State {
   abstract val searchString: String
   abstract val searchOrigin: SearchOrigin
 
-  data class Results(
+  data class Results @Deprecated("Use from()") constructor(
     override val searchString: String,
     val values: List<SearchListItem>,
     override val searchOrigin: SearchOrigin
-  ) : State()
+  ) : State() {
+    companion object {
+      fun from(
+        searchString: String,
+        values: List<SearchListItem>,
+        searchOrigin: SearchOrigin
+      ): Results {
+        val reorderedValues = if (values.none { it.value == searchString }) {
```

Comment:
I had a quick playaround with a kotlin scratch

![image](https://user-images.githubusercontent.com/3358282/80464717-1a401980-8932-11ea-8507-d32ae36c01e9.png)

sortedByDescending is the closest to what we want to do and gets bonus points for having no effect if the item is not present

Reply:
I changed it to `sortedByDescending` as you requested.

That code looks much cleaner. I never knew you could use a boolean there.